surfer/utils.py

```diff
@@ -76,20 +104,32 @@ def __init__(self, subject_id, hemi, surf, subjects_dir=None,
         self.hemi = hemi
         self.surf = surf
         self.offset = offset
+        self.coords = None
+        self.faces = None
+        self.nn = None
 
         subjects_dir = _get_subjects_dir(subjects_dir)
         self.data_path = op.join(subjects_dir, subject_id)
 
     def load_geometry(self):
         surf_path = op.join(self.data_path, "surf",
                             "%s.%s" % (self.hemi, self.surf))
-        self.coords, self.faces = nib.freesurfer.read_geometry(surf_path)
+        coords, faces = nib.freesurfer.read_geometry(surf_path)
         if self.offset is not None:
             if self.hemi == 'lh':
-                self.coords[:, 0] -= (np.max(self.coords[:, 0]) + self.offset)
+                coords[:, 0] -= (np.max(coords[:, 0]) + self.offset)
             else:
-                self.coords[:, 0] -= (np.min(self.coords[:, 0]) + self.offset)
-        self.nn = _compute_normals(self.coords, self.faces)
+                coords[:, 0] -= (np.min(coords[:, 0]) + self.offset)
+        nn = _compute_normals(coords, faces)
+
+        if self.coords is None:
+            self.coords = coords
+            self.faces = faces
+            self.nn = nn
+        else:
+            self.coords[:] = coords
+            self.faces[:] = faces
+            self.nn[:] = nn
```

Comment:
Since mesh objects now use the data stored on the `Surface` objects I changed this to modify it in place, assuming that `Surface` objects are not used independently; if you think that this is too much change I can make it optional through a parameter.